src/main/resources/templates/main.html

```diff
@@ -1,12 +1,12 @@
 <!DOCTYPE html>
-<html xmlns:th="http://www.thymeleaf.org">
+<html lang="en" xmlns:th="http://www.thymeleaf.org">
 <head>
     <meta charset="UTF-8">
     <title>Vendor Profit</title>
 </head>
 <body>
 test testing.
+<p th:text="${product}">
 <p th:text="${product}"></p>
-<p th:text="${product}" />
```

Comment:
`<p>` tag does not have a minimized form AKA `<p/>`. If you want to omit a closing tag you may do so as long it's followed by one of the tags as stated in the reference. In this case, a `<p>` tag does not need to be closed if it's followed by another `<p>` tag. [Reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/p)

Reply:
Huh... interesting.